### ISSUE TITLE
edge pooling could not match the labels of the merged node to the original node labels

### ISSUE THREAD

**Issue by neil-jinsh-public** [pyg-team/pytorch_geometric#1613]
https://github.com/rusty1s/pytorch_geometric/blob/e025bfc9f08f66b20c917994e8f47fa5a3511c95/test/nn/pool/test_edge_pool.py#L1
@rusty1s I am writing a graph convolution network based on edge pooling for supervised node classification, but there are errors in the loss calculation due to node merging. An error occurred when the label of the merged node could not match the label of the original node. How should this problem be solved?

Thank you in advance for your reply.

**Comment by rusty1s** [comment]
So you are wanting to use edge pooling for a node classification task? It seems like you need some kind of [U-Net](https://arxiv.org/abs/1905.05178) like architecture here to achieve when using pooling ops.

**Comment by neil-jinsh-public** [comment]
Thank you for your reply.
Compared with the regular pattern of image data enhancement and node reduction, the graph structure data, when using edge pooling, the number of nodes does not decrease regularly (the edge pool in the pytorch version is edge collapse based on the calculated edge score, which is ideally half the number of nodes) .I read ‘Edge contracaction Pooling for Graph Neural Networks’ , which mentioned a good combination of Edge contracaction Pooling and GCN.But that doesn’t seem to be the case. So it’s a question worth asking.

**Comment by rusty1s** [comment]
I agree, but I'm not sure what's the problem here. In case of supervised node classification, you also need to use the unpooling information provided by `EdgePooling`.

**Comment by neil-jinsh-public** [comment]
I have noticed the unpooling information you mentioned, but the unedgepool after pooling does not seem to be consistent with the information before pooling.
Thanks again for your reply.

**Comment by rusty1s** [comment]
Can you tell me what exactly is inconsistent? I can take a closer look then.

**Comment by neil-jinsh-public** [comment]
![image](https://user-images.githubusercontent.com/49297407/92561138-38396580-f2a6-11ea-85ad-5edfd670501f.png)

The number of input nodes is 2096, after two pooling, the number of nodes becomes 634, the number of features is 6.Model () produces a classified prediction of the node, data. y is the nodes' real label.I don’t know if you can understand how I described it

**Comment by rusty1s** [comment]
Aren't you using the unpooling information regarding `EdgePool`? You need to use those to convert your model back to 2096 nodes. If you show me your network architecture, I can help you further.
